src/UI/SignTransactionsModals/index.tsx

```diff
@@ -66,22 +66,35 @@ function SignTransactionsModals({
     callbackRoute,
     className
   };
+
+  const LedgerConfirm = CustomConfirmScreens?.Ledger ? (
+    <CustomConfirmScreens.Ledger {...signProps} />
+  ) : (
+    <SignWithLedgerModal {...signProps} />
+  );
+  const WalletconnectConfirm = CustomConfirmScreens?.WalletConnect ? (
```

Comment:
I would avoid doing this outside the render, this will render the components for all providers, having it inline with && will shortcircuit it before the && condition. For example, if you sign with extension, the <SignWithWalletCOnnect> will also trigger in this case, which might have unpleasant side-effects